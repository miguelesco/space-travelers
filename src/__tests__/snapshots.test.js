/* eslint-disable linebreak-style */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../components/rockets/MainRockets';
import Missions from '../components/missions/MissionsPage';
import Profile from '../components/profile/ProfilePage';
import store from '../redux/configureStore';

describe('Tests snapshots of components', () => {
  const rockets = [
    {
      id: 1,
      name: 'Falcon 1',
      description: 'rocket',
      img: 'https://s3.amazonaws.com/launchlibrary/RocketImages/falcon1_1920.jpg',
      reserved: true,
    },
  ];

  const missions = [
    {
      id: '1',
      name: 'Go Mars',
      description: 'Yes we can',
      member: false,
    },
  ];

  it('renders correctly', () => {
    const rocketsTree = renderer
      .create(
        <Provider store={store}>
          <Rockets rockets={rockets} />
        </Provider>,
      )
      .toJSON();
    expect(rocketsTree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const missionsTree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(missionsTree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const profileTree = renderer
      .create(
        <Provider store={store}>
          <Profile rockets={rockets} missions={missions} />
        </Provider>,
      )
      .toJSON();
    expect(profileTree).toMatchSnapshot();
  });
});
