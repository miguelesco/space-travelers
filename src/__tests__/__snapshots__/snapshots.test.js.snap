// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests snapshots of components renders correctly 1`] = `
<div
  className="sc-bdfBQB jNIcXP container"
>
  <div
    className="sc-gsTEea cJmrSo card"
  >
    <div
      className="sc-dlfnuX kmUnEG"
    >
      <img
        alt="rocket"
        className="card-img"
        src="https://s3.amazonaws.com/launchlibrary/RocketImages/falcon1_1920.jpg"
      />
    </div>
    <div
      className="sc-hKgJUU dNGPWG card-body"
    >
      <h2
        className="sc-eCstlR fAaXZO card-title"
      >
        Falcon 1
      </h2>
      <div
        className="sc-jSgvzq emRafE card-text"
      >
        <small
          className="sc-gKseQn cxYiQZ position-absolute badge rounded-pill "
        >
          Reserved
        </small>
        <p>
          rocket
        </p>
      </div>
      <button
        className="sc-iBPTik fDHiaM btn btn-primary"
        onClick={[Function]}
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</div>
`;

exports[`Tests snapshots of components renders correctly 2`] = `
<div
  className="mx-5 mt-4"
>
  <table
    className="table caption-top table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          className="col-1"
        >
          <strong>
            Mission
          </strong>
        </th>
        <th
          className="col-9"
        >
          <strong>
            Description
          </strong>
        </th>
        <th
          className="col-1"
        >
          <strong>
            Status
          </strong>
        </th>
        <th
          className="col-1"
        >
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Tests snapshots of components renders correctly 3`] = `
<div
  className="row mx-5 mt-4"
>
  <div
    className="col"
  >
    <h2>
      My Missions
    </h2>
    <table
      className="table table-bordered table-hover"
    >
      <tbody />
    </table>
  </div>
  <div
    className="col"
  >
    <h2>
      My Rockets
    </h2>
    <table
      className="table table-bordered table-hover"
    >
      <tbody />
    </table>
  </div>
</div>
`;
